proto/buf/compiler/v1/report.proto

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.compiler.v1;
+
+// A diagnostic report, consisting of `Diagnostics` and the `File`s they diagnose.
+message Report {
+  // A file involved in a diagnostic `Report`.
+  message File {
+    // The path to this file. Does not need to be meaningful as a file-system path.
+    string path = 1;
+
+    // The textual contents of this file.
+    bytes text = 2;
```

Comment:
I *personally* don't think it's that big of a deal, files are relatively small as far as data is concerned. However, if you feel strongly, one thing we can do is make it so that File contains fields `startOffset` and `startLine` which specify where the file has been truncated. This minimizes the amount of code that needs to change, and is something we can add in the future if this becomes an issue. WDYT?

Reply:
Yep, TODO sounds good.